tests/contract_testcases/evm/error.sol

```diff
@@ -0,0 +1,38 @@
+
+error E1(int bar, bool);
+error E2(int bar, bool foo);
+
+contract e {
+	error E3(int bar, bool foo);
+}
+
+contract c is e {
+	error E4(int bar, bool foo);
+
+	function named1() public {
+		if (1 != 0) {
+			revert E1({bar: 1, foo: true});
+		} else if (1 != 0) {
+			revert E2({bar: 1});
+		} else if (1 != 0) {
+			revert E3({bar: 1, foo: true, baz: "hey"});
+		} else if (1 != 0) {
+			revert e.E3({bar: 1, foo: true, foo: false});
+		} else {
+			revert E4({bar: 2, baz: true});
```

Comment:
If I do `revert E3({bar: 2, bar: 1.5, foo: true})`, will there be any spurious error?

Reply:
You'll get an error saying that `bar` is duplicate.

Reply:
Won't I be warned that `1.5` is not an integer?

Reply:
No, duplicates get resolved with `ResolveTo::Unknown`